internal/cache.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"github.com/tigrisdata/tigris/errors"
+	"github.com/ugorji/go/codec"
+)
+
+// NewStreamData returns a stream data for anything that we need to store in streams.
+func NewStreamData(data []byte) *StreamData {
+	return &StreamData{
+		RawData:   data,
+		CreatedAt: NewTimestamp(),
+	}
+}
+
+// NewStreamDataForChannel is used by channels as it has additional information attached to it.
+func NewStreamDataForChannel(dataType string, clientId string, socketId string, eventName string, data []byte) *StreamData {
+	return &StreamData{
+		RawData:   data,
+		ClientId:  clientId,
```

Comment:
We isolate data by clientId or we still leverage the namespace ?

Reply:
i see this is encoded in the key/table name.

Reply:
Yeah, that key encoding still remains the same i.e. in the metadata package. This is the serialized payload that we will be storing as a value.

Reply:
I have removed this construction of stream for the channel now to avoid any confusion.